kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -97,6 +98,8 @@ private[kyuubi] class EngineRef(
     case _ => user
   }
 
+  private[kyuubi] val effectiveUser: String = if (doAsEnabled) appUser else Utils.currentUser
```

Comment:
there are three user variables in this class now,

- `user`: session user
- `appUser`: for engine routing
- `effectiveUser`: for engine launching

those name are not much straightforward, maybe I need to rename them to

- `user` => `sessionUser`
- `appUser` => `routingUser`
- `effectiveUser` => `appUser`

thought?

Reply:
+1. It's better to let appUser be consistent with the user on YARN.